airflow/models/__init__.py

```diff
@@ -16,25 +16,36 @@
 # specific language governing permissions and limitations
 # under the License.
 """Airflow models"""
-from airflow.models.base import ID_LEN, Base
-from airflow.models.baseoperator import BaseOperator, BaseOperatorLink
-from airflow.models.connection import Connection
-from airflow.models.dag import DAG, DagModel, DagTag
```

Comment:
I think we have to preserve the possibility of `from airflow.models import DAG`. The concern is that many users can use this import in their DAGs...

Reply:
And if we want users to migrate to `from airflow.models.dag import DAG`, should we add pre-commit check that we are not using an import from `airflow.models`?

Reply:
> I think we have to preserve the possibility of `from airflow.models import DAG`. The concern is that many users can use this import in their DAGs...
Yep. I think it's already implemented in the follow-up commit.

> And if we want users to migrate to `from airflow.models.dag import DAG`, should we add pre-commit check that we are not using an import from `airflow.models`?
Yes. We should @msb217  -> I can add such check quickly with AST parsing after all the comments are addressed. Can be a separate PR @nuclearpinguin ? This one is already huge!